hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshot.java

```diff
@@ -82,11 +91,30 @@ public OmSnapshot(KeyManager keyManager,
     this.bucketName = bucketName;
     this.volumeName = volumeName;
     this.omMetadataManager = keyManager.getMetadataManager();
+    this.snapshotDBKey = SnapshotInfo.getTableKey(
+        volumeName, bucketName, snapshotName);
+    this.activeOMMetadataManager = ozoneManager.getMetadataManager();
   }
 
+  /**
+   * Check snapshot status in active DB snapshotInfoTable.
+   * @throws IOException FileNotFoundException when snapshot is not active.
+   */
+  private void checkSnapshotStatus() throws IOException {
```

Comment:
If we don't allow any ops with inactive snapshots, could we just block creation of metadata manager for these snapshots? . This way we'll only have to change in one place.  Not sure if we are using it elsewhere.
https://github.com/apache/ozone/blob/8202b4184e57dede01f21bc5be5563ae85d1dc5b/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java#L183

Reply:
Thx. I've updated it to use the cleaner approach. PTAL.